esiclient/v1/node_console.py

```diff
@@ -0,0 +1,60 @@
+#   Licensed under the Apache License, Version 2.0 (the "License"); you may
+#   not use this file except in compliance with the License. You may obtain
+#   a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#   WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#   License for the specific language governing permissions and limitations
+#   under the License.
+
+import json
+import logging
+import os
+
+from osc_lib.command import command
+from osc_lib import exceptions
+from osc_lib.i18n import _
+
+
+class NodeConsoleConnect(command.Command):
+    """Connect the node console"""
+
+    log = logging.getLogger(__name__ + ".NodeConsoleConnect")
+
+    def get_parser(self, prog_name):
+        parser = super(NodeConsoleConnect, self).get_parser(prog_name)
+        parser.add_argument(
+            "node",
+            metavar="<node>",
+            help=_("node"))
+
+        return parser
+
+    def take_action(self, parsed_args):
+        self.log.debug("take_action(%s)", parsed_args)
+
+        node = parsed_args.node
+
+        ironic_client = self.app.client_manager.baremetal
+
+        try:
+            output = ironic_client.node.get_console(node)
```

Comment:
Is the try/except block needed here? If the node doesn't exist, is the output simply `None`, or does it throw an Exception? If the first, you don't need the try/except block; if the latter, then you probably don't want the if statement beneath.

Reply:
"output" does not catch any value because that function is simply throwing the error not raising the error which means rest of the lines will not be executed. if statement is just for the test case scenario because we assigned none as a return value for the unknown node test case.

Reply:
Okay - we do want to catch the exact Error then; if you run the command with `--debug` you should see it.

We shouldn't have the if statement for the test case scenario. You can have the mock console command throw an error using `side_effect` - https://docs.python.org/3/library/unittest.mock.html

Reply:
I know the exception already that's not the issue.